File: library/src/main/java/renetik/android/view/CSAutoLoadController.java
package renetik.android.view;

import renetik.android.lang.CSWork;
import renetik.android.rpc.CSResponse;
import renetik.android.viewbase.CSViewController;
import renetik.android.java.callback.CSReturn;

import static renetik.android.lang.CSLang.MINUTE;
import static renetik.android.lang.CSLang.schedule;

/**
 * Created by renetik on 29/12/17.
 */

public class CSAutoLoadController<RowType> extends CSViewController {

    private final CSWork _autoReload;

    public CSAutoLoadController(CSViewController parent, CSReturn<CSResponse> runWith) {
        super(parent);
        _autoReload = schedule(MINUTE, runWith::invoke);
    }

    protected void onViewShowing() {
        super.onViewShowing();
        _autoReload.start();
    }

    protected void onViewHiding() {
        super.onViewHiding();
        _autoReload.stop();
    }

}

File: library/src/main/java/renetik/android/view/CSAutoLoadController.kt
package renetik.android.view

import android.view.View
import renetik.android.java.callback.CSReturn
import renetik.android.lang.CSLang.MINUTE
import renetik.android.lang.CSLang.schedule
import renetik.android.rpc.CSResponse
import renetik.android.viewbase.CSViewController

/**
 * Created by renetik on 29/12/17.
 */

class CSAutoLoadController(parent: CSViewController<*>, runWith: CSReturn<CSResponse<*>>)
    : CSViewController<View>(parent) {

    private val autoReload = schedule(MINUTE, { runWith.invoke() })

    override fun onViewShowing() {
        super.onViewShowing()
        autoReload.start()
    }

    override fun onViewHiding() {
        super.onViewHiding()
        autoReload.stop()
    }

}
